test/unit/taxonomy/publishing_api_adapter_test.rb

```diff
@@ -6,72 +6,66 @@ def subject
   end
 
   test "#published_taxon_data" do
-    setup_published_taxons([published_taxon])
+    setup_taxons([visible_published_taxon, not_visible_published_taxon], with_drafts: false)
+    setup_taxons([visible_draft_taxon, not_visible_draft_taxon, visible_published_taxon, not_visible_published_taxon], with_drafts: true)
     result = subject.published_taxon_data
-    assert_equal result, [published_taxon_tree]
+    assert_same_elements result, [taxon_tree(visible_published_taxon), taxon_tree(not_visible_published_taxon)]
   end
 
   test "#draft_taxon_data" do
-    setup_published_taxons([published_taxon])
-    setup_draft_taxons([visible_draft_taxon, draft_taxon])
+    setup_taxons([visible_published_taxon, not_visible_published_taxon], with_drafts: false)
+    setup_taxons([visible_draft_taxon, not_visible_draft_taxon, visible_published_taxon, not_visible_published_taxon], with_drafts: true)
     result = subject.draft_taxon_data
-    assert_equal [visible_draft_taxon_tree], result
+    assert_same_elements [taxon_tree(visible_draft_taxon)], result
   end
 
-  def published_taxon_tree
-    published_taxon.tap do |taxon|
-      taxon['expanded_links_hash'] = published_taxon
+  def taxon_tree(taxon)
+    taxon.tap do |t|
+      t['expanded_links_hash'] = taxon.dup
     end
   end
 
-  def published_taxon
-    { "content_id" => "published" }
+  def visible_published_taxon
+    { "content_id" => "published_visible",
+      "details" => {
+        "visible_to_departmental_editors" => true
+      } }
```

Comment:
I'd prefer if the indentation of this was like other parts of the file, even if just for consistency https://github.com/alphagov/whitehall/blob/e0328f2497a59d8b0e8b4ad7a5fb4e21c1ca880c/test/unit/taxonomy/publishing_api_adapter_test.rb#L50-L55